indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/PartitionStat.java

```diff
@@ -40,79 +39,72 @@
 
   // Primary partition key
   private final Interval interval;
-  // Secondary partition key
-  private final int partitionId;
 
   // numRows and sizeBytes are always null currently and will be filled properly in the future.
   @Nullable
   private final Integer numRows;
   @Nullable
   private final Long sizeBytes;
 
-  @JsonCreator
-  public PartitionStat(
-      @JsonProperty("taskExecutorHost") String taskExecutorHost,
-      @JsonProperty("taskExecutorPort") int taskExecutorPort,
-      @JsonProperty("useHttps") boolean useHttps,
-      @JsonProperty("interval") Interval interval,
-      @JsonProperty("partitionId") int partitionId,
-      @JsonProperty("numRows") @Nullable Integer numRows,
-      @JsonProperty("sizeBytes") @Nullable Long sizeBytes
+  PartitionStat(
+      String taskExecutorHost,
+      int taskExecutorPort,
+      boolean useHttps,
+      Interval interval,
+      @Nullable Integer numRows,
+      @Nullable Long sizeBytes
   )
   {
     this.taskExecutorHost = taskExecutorHost;
     this.taskExecutorPort = taskExecutorPort;
     this.useHttps = useHttps;
     this.interval = interval;
-    this.partitionId = partitionId;
     this.numRows = numRows == null ? 0 : numRows;
     this.sizeBytes = sizeBytes == null ? 0 : sizeBytes;
   }
 
   @JsonProperty
-  public String getTaskExecutorHost()
+  public final String getTaskExecutorHost()
   {
     return taskExecutorHost;
   }
 
   @JsonProperty
-  public int getTaskExecutorPort()
+  public final int getTaskExecutorPort()
   {
     return taskExecutorPort;
   }
 
   @JsonProperty
-  public boolean isUseHttps()
+  public final boolean isUseHttps()
   {
     return useHttps;
   }
 
   @JsonProperty
-  public Interval getInterval()
+  public final Interval getInterval()
   {
     return interval;
   }
 
-  @JsonProperty
-  public int getPartitionId()
-  {
-    return partitionId;
-  }
-
   @Nullable
   @JsonProperty
-  public Integer getNumRows()
+  public final Integer getNumRows()
   {
     return numRows;
   }
 
   @Nullable
   @JsonProperty
-  public Long getSizeBytes()
+  public final Long getSizeBytes()
   {
     return sizeBytes;
   }
 
+  abstract int getPartitionId();
+
+  abstract T getSecondaryPartition();
+
```

Comment:
super nit: javadocs - these seem like obvious things that someone more familiar with the codebase will know just from the name. Feel free to ignore.